app/controllers/register_controller.rb

```diff
@@ -1,75 +1,108 @@
 class RegisterController < ApplicationController
 
   include ApplicationHelper
-  helper_method :get_all_attribute_names_prettified, :prepare_register_name
+  helper_method :prepare_register_name, :get_description_for_register_field
 
-  # def index(register)
-  #   @countries = OpenRegister.register(register, :beta)._all_records
-  # end
+  @register_phase = Settings.register_phase
+
+  before_action :confirm, only: [:create]
 
   def index
-    @register_name = prepare_register_name(params[:register])
-    @register = OpenRegister.register(params[:register].downcase, :beta)._all_records.sort_by{|register| register.name}
+    @register = get_register(params[:register])._all_records
+    @register[0].try(:name) ? @register = @register.sort_by(&:name) : @register = @register.sort_by(&:key)
   end
 
-  def show
+  def new
+    @register = get_register(params[:register])
+    @form = JSON.parse(params.to_json)
+    render 'form'
```

Comment:
Yes, I agree following conversation, we should use patterns over efficiency in this case. Good call.